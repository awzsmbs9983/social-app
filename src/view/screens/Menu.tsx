import React, {useEffect} from 'react'
import {
  StyleProp,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native'
import VersionNumber from 'react-native-version-number'
import {s, colors} from '../lib/styles'
import {ScreenParams} from '../routes'
import {useStores} from '../../state'
import {
  HomeIcon,
  UserGroupIcon,
  BellIcon,
  CogIcon,
  MagnifyingGlassIcon,
} from '../lib/icons'
import {UserAvatar} from '../com/util/UserAvatar'
import {ViewHeader} from '../com/util/ViewHeader'
import {CreateSceneModel} from '../../state/models/shell-ui'

export const Menu = ({navIdx, visible}: ScreenParams) => {
  const store = useStores()

  useEffect(() => {
    if (visible) {
      store.nav.setTitle(navIdx, 'Menu')
      // trigger a refresh in case memberships have changed recently
      store.me.refreshMemberships()
    }
  }, [store, visible])

  // events
  // =

  const onNavigate = (url: string) => {
    if (url === '/notifications') {
      store.nav.switchTo(1, true)
    } else {
      store.nav.switchTo(0, true)
      store.nav.navigate(url)
    }
  }
  const onPressCreateScene = () => {
    store.shell.openModal(new CreateSceneModel())
  }

  // rendering
  // =

  const MenuItem = ({
    icon,
    label,
    count,
    url,
    bold,
    onPress,
  }: {
    icon: JSX.Element
    label: string
    count?: number
    url?: string
    bold?: boolean
    onPress?: () => void
  }) => (
    <TouchableOpacity
      style={styles.menuItem}
      onPress={onPress ? onPress : () => onNavigate(url || '/')}>
      <View style={[styles.menuItemIconWrapper]}>
        {icon}
        {count ? (
          <View style={styles.menuItemCount}>
            <Text style={styles.menuItemCountLabel}>{count}</Text>
          </View>
        ) : undefined}
      </View>
      <Text
        style={[
          styles.menuItemLabel,
          bold ? styles.menuItemLabelBold : undefined,
        ]}
        numberOfLines={1}>
        {label}
      </Text>
    </TouchableOpacity>
  )

  /*TODO <MenuItem icon={['far', 'compass']} label="Discover" url="/" />*/
  return (
    <View style={styles.view}>
      <ViewHeader title="Bluesky" subtitle="Private Beta" />
      <TouchableOpacity
        style={styles.searchBtn}
        onPress={() => onNavigate('/search')}>
        <MagnifyingGlassIcon
          style={{color: colors.gray5} as StyleProp<ViewStyle>}
          size={21}
        />
        <Text style={styles.searchBtnLabel}>Search</Text>
      </TouchableOpacity>
      <View style={styles.section}>
        <MenuItem
          icon={
            <UserAvatar
              size={24}
              displayName={store.me.displayName}
              handle={store.me.handle}
              avatar={store.me.avatar}
            />
          }
          label={store.me.displayName || store.me.handle}
          bold
          url={`/profile/${store.me.handle}`}
        />
        <MenuItem
          icon={
            <HomeIcon
              style={{color: colors.gray5} as StyleProp<ViewStyle>}
              size="24"
            />
          }
          label="Home"
          url="/"
        />
        <MenuItem
          icon={
            <BellIcon
              style={{color: colors.gray5} as StyleProp<ViewStyle>}
              size="24"
            />
          }
          label="Notifications"
          url="/notifications"
          count={store.me.notificationCount}
        />
        <MenuItem
          icon={
            <CogIcon
              style={{color: colors.gray6} as StyleProp<ViewStyle>}
              size="24"
              strokeWidth={2}
            />
          }
          label="Settings"
          url="/settings"
        />
      </View>
      <View style={styles.section}>
        <Text style={styles.heading}>Scenes</Text>
        <MenuItem
          icon={
            <UserGroupIcon
              style={{color: colors.gray6} as StyleProp<ViewStyle>}
              size="24"
            />
          }
          label="Create a scene"
          onPress={onPressCreateScene}
        />
        {store.me.memberships
          ? store.me.memberships.memberships.map((membership, i) => (
              <MenuItem
                key={i}
                icon={
                  <UserAvatar
                    size={24}
                    displayName={membership.displayName}
                    handle={membership.handle}
                    avatar={membership.avatar}
                  />
                }
                label={membership.displayName || membership.handle}
                url={`/profile/${membership.handle}`}
              />
            ))
          : undefined}
      </View>
      <View style={styles.footer}>
        <Text style={s.gray4}>
          Build version {VersionNumber.appVersion} ({VersionNumber.buildVersion}
          )
        </Text>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  view: {
    flex: 1,
    backgroundColor: colors.white,
  },
  section: {
    paddingHorizontal: 10,
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 1,
    borderBottomColor: colors.gray1,
  },
  heading: {
    fontSize: 16,
    fontWeight: 'bold',
    paddingVertical: 8,
    paddingHorizontal: 4,
  },

  searchBtn: {
    flexDirection: 'row',
    backgroundColor: colors.gray1,
    borderRadius: 8,
    margin: 10,
    marginBottom: 0,
    paddingVertical: 10,
    paddingHorizontal: 12,
  },
  searchBtnLabel: {
    marginLeft: 8,
    fontSize: 18,
    color: colors.gray6,
  },

  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
    paddingHorizontal: 2,
  },
  menuItemIconWrapper: {
    width: 30,
    height: 30,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 10,
  },
  menuItemLabel: {
    fontSize: 17,
    color: colors.gray7,
  },
  menuItemLabelBold: {
    fontWeight: 'bold',
  },
  menuItemCount: {
    position: 'absolute',
    right: -6,
    top: -2,
    backgroundColor: colors.red3,
    paddingHorizontal: 4,
    paddingBottom: 1,
    borderRadius: 6,
  },
  menuItemCountLabel: {
    fontSize: 12,
    fontWeight: 'bold',
    color: colors.white,
  },

  footer: {
    paddingHorizontal: 14,
    paddingVertical: 18,
  },
})
