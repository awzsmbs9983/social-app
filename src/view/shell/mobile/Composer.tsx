import React, {useEffect} from 'react'
import {observer} from 'mobx-react-lite'
import {Animated, Easing, Platform, StyleSheet, View} from 'react-native'
import {ComposePost} from '../../com/composer/ComposePost'
import {ComposerOpts} from '../../../state/models/shell-ui'
import {useAnimatedValue} from '../../lib/hooks/useAnimatedValue'

export const Composer = observer(
  ({
    active,
    winHeight,
    replyTo,
    onPost,
    onClose,
  }: {
    active: boolean
    winHeight: number
    replyTo?: ComposerOpts['replyTo']
    onPost?: ComposerOpts['onPost']
    onClose: () => void
  }) => {
    const initInterp = useAnimatedValue(0)

    useEffect(() => {
      if (active) {
        Animated.timing(initInterp, {
          toValue: 1,
          duration: 300,
          easing: Easing.out(Easing.exp),
          useNativeDriver: true,
        }).start()
      } else {
        initInterp.setValue(0)
      }
    }, [initInterp, active])
    const wrapperAnimStyle = {
      transform: [
        {
          translateY: initInterp.interpolate({
            inputRange: [0, 1],
            outputRange: [winHeight, 0],
          }),
        },
      ],
    }

    // events
    // =

    // rendering
    // =

    if (!active) {
      return <View />
    }

    return (
      <Animated.View style={[styles.wrapper, wrapperAnimStyle]}>
        <ComposePost replyTo={replyTo} onPost={onPost} onClose={onClose} />
      </Animated.View>
    )
  },
)

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    width: '100%',
    backgroundColor: '#fff',
    ...Platform.select({
      ios: {
        paddingTop: 24,
      },
    }),
  },
})
