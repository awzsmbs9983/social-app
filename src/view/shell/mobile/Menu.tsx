import React, {useEffect} from 'react'
import {
  ScrollView,
  StyleProp,
  StyleSheet,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native'
import {observer} from 'mobx-react-lite'
import VersionNumber from 'react-native-version-number'
import {s, colors} from '../../lib/styles'
import {useStores} from '../../../state'
import {
  HomeIcon,
  UserGroupIcon,
  BellIcon,
  CogIcon,
  MagnifyingGlassIcon,
} from '../../lib/icons'
import {UserAvatar} from '../../com/util/UserAvatar'
import {Text} from '../../com/util/text/Text'
import {ToggleButton} from '../../com/util/forms/ToggleButton'
import {CreateSceneModal} from '../../../state/models/shell-ui'
import {usePalette} from '../../lib/hooks/usePalette'

export const Menu = observer(
  ({visible, onClose}: {visible: boolean; onClose: () => void}) => {
    const pal = usePalette('default')
    const store = useStores()

    useEffect(() => {
      if (visible) {
        // trigger a refresh in case memberships have changed recently
        // TODO this impacts performance, need to find the right time to do this
        // store.me.refreshMemberships()
      }
    }, [store, visible])

    // events
    // =

    const onNavigate = (url: string) => {
      onClose()
      if (url === '/notifications') {
        store.nav.switchTo(1, true)
      } else {
        store.nav.switchTo(0, true)
        if (url !== '/') {
          store.nav.navigate(url)
        }
      }
    }
    const onPressCreateScene = () => {
      onClose()
      store.shell.openModal(new CreateSceneModal())
    }

    // rendering
    // =

    const MenuItem = ({
      icon,
      label,
      count,
      url,
      bold,
      onPress,
    }: {
      icon: JSX.Element
      label: string
      count?: number
      url?: string
      bold?: boolean
      onPress?: () => void
    }) => (
      <TouchableOpacity
        style={styles.menuItem}
        onPress={onPress ? onPress : () => onNavigate(url || '/')}>
        <View style={[styles.menuItemIconWrapper]}>
          {icon}
          {count ? (
            <View style={styles.menuItemCount}>
              <Text style={styles.menuItemCountLabel}>{count}</Text>
            </View>
          ) : undefined}
        </View>
        <Text
          type="h4"
          style={[
            pal.text,
            bold ? styles.menuItemLabelBold : styles.menuItemLabel,
          ]}
          numberOfLines={1}>
          {label}
        </Text>
      </TouchableOpacity>
    )

    return (
      <ScrollView style={[styles.view, pal.view]}>
        <TouchableOpacity
          onPress={() => onNavigate(`/profile/${store.me.handle}`)}
          style={styles.profileCard}>
          <UserAvatar
            size={60}
            displayName={store.me.displayName}
            handle={store.me.handle}
            avatar={store.me.avatar}
          />
          <View style={s.flex1}>
            <Text
              type="h3"
              style={[pal.text, styles.profileCardDisplayName]}
              numberOfLines={1}>
              {store.me.displayName || store.me.handle}
            </Text>
            <Text
              style={[pal.textLight, styles.profileCardHandle]}
              numberOfLines={1}>
              @{store.me.handle}
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.searchBtn, pal.btn]}
          onPress={() => onNavigate('/search')}>
          <MagnifyingGlassIcon
            style={pal.text as StyleProp<ViewStyle>}
            size={25}
          />
          <Text type="h4" style={[pal.text, styles.searchBtnLabel]}>
            Search
          </Text>
        </TouchableOpacity>
        <View style={[styles.section, pal.border]}>
          <MenuItem
            icon={
              <HomeIcon style={pal.text as StyleProp<ViewStyle>} size="26" />
            }
            label="Home"
            url="/"
          />
          <MenuItem
            icon={
              <BellIcon style={pal.text as StyleProp<ViewStyle>} size="28" />
            }
            label="Notifications"
            url="/notifications"
            count={store.me.notificationCount}
          />
        </View>
        <View style={[styles.section, pal.border]}>
          <Text type="h5" style={[pal.text, styles.heading]}>
            Scenes
          </Text>
          {store.me.memberships
            ? store.me.memberships.memberships.map((membership, i) => (
                <MenuItem
                  key={i}
                  icon={
                    <UserAvatar
                      size={34}
                      displayName={membership.displayName}
                      handle={membership.handle}
                      avatar={membership.avatar}
                    />
                  }
                  label={membership.displayName || membership.handle}
                  url={`/profile/${membership.handle}`}
                />
              ))
            : undefined}
        </View>
        <View style={[styles.section, pal.border]}>
          <MenuItem
            icon={
              <UserGroupIcon
                style={pal.text as StyleProp<ViewStyle>}
                size="30"
              />
            }
            label="Create a scene"
            onPress={onPressCreateScene}
          />
          <MenuItem
            icon={
              <CogIcon
                style={pal.text as StyleProp<ViewStyle>}
                size="30"
                strokeWidth={2}
              />
            }
            label="Settings"
            url="/settings"
          />
        </View>
        <View style={[styles.section, pal.border]}>
          <ToggleButton
            label="Dark mode"
            isSelected={store.shell.darkMode}
            onPress={() => store.shell.setDarkMode(!store.shell.darkMode)}
          />
        </View>
        <View style={styles.footer}>
          <Text style={[pal.textLight]}>
            Build version {VersionNumber.appVersion} (
            {VersionNumber.buildVersion})
          </Text>
        </View>
        <View style={{height: 100}} />
      </ScrollView>
    )
  },
)

const styles = StyleSheet.create({
  view: {
    flex: 1,
  },
  section: {
    paddingHorizontal: 10,
    paddingTop: 10,
    paddingBottom: 10,
    borderBottomWidth: 1,
  },
  heading: {
    paddingVertical: 8,
    paddingHorizontal: 4,
  },

  profileCard: {
    flexDirection: 'row',
    alignItems: 'center',
    margin: 10,
    marginBottom: 6,
  },
  profileCardDisplayName: {
    marginLeft: 12,
  },
  profileCardHandle: {
    marginLeft: 12,
  },

  searchBtn: {
    flexDirection: 'row',
    borderRadius: 8,
    margin: 10,
    marginBottom: 0,
    paddingVertical: 10,
    paddingHorizontal: 12,
  },
  searchBtnLabel: {
    marginLeft: 14,
    fontWeight: 'normal',
  },

  menuItem: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingVertical: 8,
    paddingLeft: 6,
    paddingRight: 10,
  },
  menuItemIconWrapper: {
    width: 36,
    height: 36,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 12,
  },
  menuItemLabel: {
    flex: 1,
    fontWeight: 'normal',
  },
  menuItemLabelBold: {
    flex: 1,
    fontWeight: 'bold',
  },
  menuItemCount: {
    position: 'absolute',
    right: -6,
    top: -2,
    backgroundColor: colors.red3,
    paddingHorizontal: 4,
    paddingBottom: 1,
    borderRadius: 6,
  },
  menuItemCountLabel: {
    fontSize: 12,
    fontWeight: 'bold',
    color: colors.white,
  },

  footer: {
    paddingHorizontal: 14,
    paddingVertical: 18,
  },
})
