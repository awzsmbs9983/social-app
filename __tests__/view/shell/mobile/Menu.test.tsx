import React from 'react'
import {Menu} from '../../../../src/view/shell/mobile/Menu'
import renderer from 'react-test-renderer'
// import {render} from '../../../../jest/test-utils'

describe('Menu', () => {
  const mockedProps = {
    visible: true,
    onClose: jest.fn(),
  }
  it('renders correctly', () => {
    const tree = renderer.create(<Menu {...mockedProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
