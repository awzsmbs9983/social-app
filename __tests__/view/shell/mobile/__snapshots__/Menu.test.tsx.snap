// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
        "marginBottom": 6,
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={60}
      bbWidth={60}
      focusable={false}
      height={60}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 60,
            "width": 60,
          },
        ]
      }
      vbHeight={100}
      vbWidth={100}
      width={60}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1292135,
                1,
                -2424577,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="grad"
            x1="0"
            x2="1"
            y1="0"
            y2="1"
          />
        </RNSVGDefs>
        <RNSVGCircle
          cx="50"
          cy="50"
          fill={
            Array [
              1,
              "grad",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
          r="50"
        />
        <RNSVGText
          content={null}
          dx={Array []}
          dy={Array []}
          fill={4294967295}
          font={
            Object {
              "fontSize": "50",
              "fontWeight": "bold",
              "textAnchor": "middle",
            }
          }
          propList={
            Array [
              "fill",
            ]
          }
          rotate={Array []}
          x={
            Array [
              "50",
            ]
          }
          y={
            Array [
              "67",
            ]
          }
        >
          <RNSVGTSpan
            content="X"
            dx={Array []}
            dy={Array []}
            font={Object {}}
            rotate={Array []}
            x={Array []}
            y={Array []}
          />
        </RNSVGText>
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "color": "#2D2626",
              "fontSize": 24,
              "fontWeight": "bold",
              "marginLeft": 12,
            },
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "color": "#423737",
              "fontSize": 18,
              "marginLeft": 12,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f8f3f3",
        "borderRadius": 8,
        "flexDirection": "row",
        "margin": 10,
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 12,
        "paddingVertical": 10,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={25}
      bbWidth={25}
      color={4284765268}
      fill="none"
      focusable={false}
      height={25}
      meetOrSlice={0}
      minX={0}
      minY={0}
      stroke="currentColor"
      strokeWidth={2}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#645454",
          },
          Object {
            "flex": 0,
            "height": 25,
            "width": 25,
          },
        ]
      }
      tintColor={4284765268}
      vbHeight={24}
      vbWidth={24}
      width={25}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={
          Array [
            2,
          ]
        }
        strokeWidth={2}
      >
        <RNSVGPath
          d="M21 21l-5.197-5.197m0 0A7.5 7.5 0 105.196 5.196a7.5 7.5 0 0010.607 10.607z"
          propList={
            Array [
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          strokeLinecap={1}
          strokeLinejoin={1}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "color": "#423737",
            "fontSize": 19,
            "marginLeft": 8,
          },
        ]
      }
    >
      Search
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#f8f3f3",
        "borderBottomWidth": 1,
        "paddingBottom": 10,
        "paddingHorizontal": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 12,
              "width": 36,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="26"
          bbWidth="26"
          color={4282529591}
          focusable={false}
          height="26"
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#423737",
              },
              Object {
                "flex": 0,
                "height": 26,
                "width": 26,
              },
            ]
          }
          tintColor={4282529591}
          vbHeight={48}
          vbWidth={48}
          width="26"
        >
          <RNSVGGroup
            propList={
              Array [
                "stroke",
              ]
            }
            stroke={
              Array [
                2,
              ]
            }
          >
            <RNSVGPath
              d="M 23.951 2 C 23.631 2.011 23.323 2.124 23.072 2.322 L 8.859 13.52 C 7.055 14.941 6 17.114 6 19.41 L 6 38.5 C 6 39.864 7.136 41 8.5 41 L 18.5 41 C 19.864 41 21 39.864 21 38.5 L 21 28.5 C 21 28.205 21.205 28 21.5 28 L 26.5 28 C 26.795 28 27 28.205 27 28.5 L 27 38.5 C 27 39.864 28.136 41 29.5 41 L 39.5 41 C 40.864 41 42 39.864 42 38.5 L 42 19.41 C 42 17.114 40.945 14.941 39.141 13.52 L 24.928 2.322 C 24.65 2.103 24.304 1.989 23.951 2 Z"
              propList={
                Array [
                  "strokeWidth",
                ]
              }
              strokeWidth={4}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "color": "#2D2626",
                "fontSize": 19,
              },
              undefined,
            ],
          ]
        }
      >
        Home
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 12,
              "width": 36,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="28"
          bbWidth="28"
          color={4282529591}
          fill="none"
          focusable={false}
          height="28"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#423737",
              },
              Object {
                "flex": 0,
                "height": 28,
                "width": 28,
              },
            ]
          }
          tintColor={4282529591}
          vbHeight={24}
          vbWidth={24}
          width="28"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M0 0h24v24H0z"
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M20 17h2v2H2v-2h2v-7a8 8 0 1 1 16 0v7zm-2 0v-7a6 6 0 1 0-12 0v7h12zm-9 4h6v2H9v-2z"
              fill={
                Array [
                  2,
                ]
              }
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "color": "#2D2626",
                "fontSize": 19,
              },
              undefined,
            ],
          ]
        }
      >
        Notifications
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#f8f3f3",
        "borderBottomWidth": 1,
        "paddingBottom": 10,
        "paddingHorizontal": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "paddingHorizontal": 4,
            "paddingVertical": 8,
          },
        ]
      }
    >
      Scenes
    </Text>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#f8f3f3",
        "borderBottomWidth": 1,
        "paddingBottom": 10,
        "paddingHorizontal": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 12,
              "width": 36,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="30"
          bbWidth="30"
          color={4282529591}
          fill="none"
          focusable={false}
          height="30"
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeWidth={2}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#423737",
              },
              Object {
                "flex": 0,
                "height": 30,
                "width": 30,
              },
            ]
          }
          tintColor={4282529591}
          vbHeight={24}
          vbWidth={24}
          width="30"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              Array [
                2,
              ]
            }
            strokeWidth={2}
          >
            <RNSVGPath
              d="M18 18.72a9.094 9.094 0 003.741-.479 3 3 0 00-4.682-2.72m.94 3.198l.001.031c0 .225-.012.447-.037.666A11.944 11.944 0 0112 21c-2.17 0-4.207-.576-5.963-1.584A6.062 6.062 0 016 18.719m12 0a5.971 5.971 0 00-.941-3.197m0 0A5.995 5.995 0 0012 12.75a5.995 5.995 0 00-5.058 2.772m0 0a3 3 0 00-4.681 2.72 8.986 8.986 0 003.74.477m.94-3.197a5.971 5.971 0 00-.94 3.197M15 6.75a3 3 0 11-6 0 3 3 0 016 0zm6 3a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0zm-13.5 0a2.25 2.25 0 11-4.5 0 2.25 2.25 0 014.5 0z"
              propList={
                Array [
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              strokeLinecap={1}
              strokeLinejoin={1}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "color": "#2D2626",
                "fontSize": 19,
              },
              undefined,
            ],
          ]
        }
      >
        Create a scene
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "opacity": 1,
          "paddingHorizontal": 6,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 36,
              "justifyContent": "center",
              "marginRight": 12,
              "width": 36,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="30"
          bbWidth="30"
          color={4282529591}
          fill="none"
          focusable={false}
          height="30"
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeWidth={2}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "color": "#423737",
              },
              Object {
                "flex": 0,
                "height": 30,
                "width": 30,
              },
            ]
          }
          tintColor={4282529591}
          vbHeight={24}
          vbWidth={24}
          width="30"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={
              Array [
                2,
              ]
            }
            strokeWidth={2}
          >
            <RNSVGPath
              d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.324.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 011.37.49l1.296 2.247a1.125 1.125 0 01-.26 1.431l-1.003.827c-.293.24-.438.613-.431.992a6.759 6.759 0 010 .255c-.007.378.138.75.43.99l1.005.828c.424.35.534.954.26 1.43l-1.298 2.247a1.125 1.125 0 01-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.57 6.57 0 01-.22.128c-.331.183-.581.495-.644.869l-.213 1.28c-.09.543-.56.941-1.11.941h-2.594c-.55 0-1.02-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 01-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 01-1.369-.49l-1.297-2.247a1.125 1.125 0 01.26-1.431l1.004-.827c.292-.24.437-.613.43-.992a6.932 6.932 0 010-.255c.007-.378-.138-.75-.43-.99l-1.004-.828a1.125 1.125 0 01-.26-1.43l1.297-2.247a1.125 1.125 0 011.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.087.22-.128.332-.183.582-.495.644-.869l.214-1.281z"
              propList={
                Array [
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              strokeLinecap={1}
              strokeLinejoin={1}
            />
            <RNSVGPath
              d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              propList={
                Array [
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              strokeLinecap={1}
              strokeLinejoin={1}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Array [
              Object {
                "color": "#2D2626",
                "fontSize": 19,
              },
              undefined,
            ],
          ]
        }
      >
        Settings
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 14,
        "paddingVertical": 18,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
          },
          Object {
            "color": "#968d8d",
          },
        ]
      }
    >
      Build version 
       (
      )
    </Text>
  </View>
</View>
`;
